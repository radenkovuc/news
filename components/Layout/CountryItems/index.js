import styled from 'styled-components'
import React from 'react'
import CountryItem from './CountryItem'
import withContext from '../../HOCs/withContext'
import {COUNTRIES} from '../../../common/consts.json'

const Wrapper = styled.div`
  display: flex;
  align-self: center;
  margin: 0;
  width: 100%;
  border-bottom: 1px solid #444;

  @media only screen and (min-width: 600px) {
    width: auto;
  }
`

const CountryItems = (props) => {
  const {
    isDisabled,
    appContext: {selectedCountry, setContextData}
  } = props

  const countries = []

  // eslint-disable-next-line no-unused-vars
  for (let key in COUNTRIES) {
    // eslint-disable-next-line no-prototype-builtins
    if (COUNTRIES.hasOwnProperty(key)) {
      const {langKey} = COUNTRIES[key]
      countries.push(
        <CountryItem
          key={key}
          text={langKey}
          isActive={selectedCountry === key}
          isDisabled={isDisabled}
          onClick={() => !isDisabled && setContextData({selectedCountry: key})}
        />
      )
    }
  }

  return <Wrapper>{countries}</Wrapper>
}

export default withContext(CountryItems)
