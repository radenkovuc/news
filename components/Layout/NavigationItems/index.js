import {HEADER_MENU_ITEMS as headerRoutes} from '../../../common/consts.json'
import styled from 'styled-components'
import React from 'react'
import NavigationItem from './NavigationItem'
import {withRouter} from 'next/router'

const Wrapper = styled.div`
  display: flex;
  flex: 1;
  align-self: center;
  margin: 0;
`

const NavigationItems = (props) => {
  const navItems = []

  // eslint-disable-next-line no-unused-vars
  for (let route in headerRoutes) {
    // eslint-disable-next-line no-prototype-builtins
    if (headerRoutes.hasOwnProperty(route)) {
      navItems.push(
        <NavigationItem
          key={headerRoutes[route].link}
          link={headerRoutes[route].link}
          text={headerRoutes[route].text}
          isActive={props.router && props.router.route === headerRoutes[route].link}
        />
      )
    }
  }

  return <Wrapper>{navItems}</Wrapper>
}

export default withRouter(NavigationItems)
