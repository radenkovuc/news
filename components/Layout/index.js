import styled from 'styled-components'
import React from 'react'
import NavigationItems from './NavigationItems'
import CountryItems from './CountryItems'

const Container = styled.div`
  border: 1px solid #444;
  min-height: 95vh;
  margin: 20px;
  display: flex;
  flex-direction: column;
`
const MenuContainer = styled.div`
  display: flex;
  flex-direction: column;

  @media only screen and (min-width: 600px) {
    flex-direction: row;
  }
`

type Props = {
  children: React.Component,
  disableSelectionCountry: Boolean
}

const Layout = (props: Props) => {
  const {children, disableSelectionCountry} = props

  const renderMenu = () => {
    return (
      <MenuContainer>
        <NavigationItems />
        <CountryItems isDisabled={disableSelectionCountry} />
      </MenuContainer>
    )
  }

  return (
    <Container>
      {renderMenu()}
      {children}
    </Container>
  )
}

Layout.defaultProps = {
  children: null,
  disableSelectionCountry: false
}

export default Layout
