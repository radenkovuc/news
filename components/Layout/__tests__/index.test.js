import React from 'react'
import renderer from 'react-test-renderer'
import Layout from '../index'

describe('<Layout />', () => {
  it('match snapshot', () => {
    const tree = renderer.create(<Layout />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
