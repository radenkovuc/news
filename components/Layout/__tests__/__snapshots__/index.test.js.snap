// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> match snapshot 1`] = `
<div
  className="Container-t0lxgb-0 grHKXg"
>
  <div
    className="MenuContainer-t0lxgb-1 eDixhJ"
  >
    <div
      className="Wrapper-sc-3didjk-0 ghaWSH"
    >
      <div
        className="Item-sc-14w5znm-0 flAooz"
      >
        <span
          className="Text-sc-14w5znm-1 cMAzLR"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          TOP_NEWS
        </span>
      </div>
      <div
        className="Item-sc-14w5znm-0 flAooz"
      >
        <span
          className="Text-sc-14w5znm-1 cMAzLR"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          CATEGORIES
        </span>
      </div>
      <div
        className="Item-sc-14w5znm-0 flAooz"
      >
        <span
          className="Text-sc-14w5znm-1 cMAzLR"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          SEARCH
        </span>
      </div>
    </div>
    <div
      className="Wrapper-sc-176nf4l-0 hSHBvl"
    >
      <div
        className="Item-fwh4iu-0 bNiPnD"
        onClick={[Function]}
      >
        <span
          className="Text-fwh4iu-1 KCUNf"
        >
          GB
        </span>
      </div>
      <div
        className="Item-fwh4iu-0 bNiPnD"
        onClick={[Function]}
      >
        <span
          className="Text-fwh4iu-1 KCUNf"
        >
          US
        </span>
      </div>
    </div>
  </div>
</div>
`;
